#! /usr/bin/env python


# Initialize ros and import parameter generator
PACKAGE='motion_force_control'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


# Add parameters
#       Name     Type    Level Description     	Default 	Min   	Max
gen.add("convergenceRate",double_t,0,"Convergence scale",5.0,1.0,10.0)
gen.add("filteredForceGain",double_t,0,"filtering weight of force torque sensor",0.9,0,1)
gen.add("contactForceThreshold",double_t,0,"Force threshold used to detect contact [N]",6,0,10)
gen.add("controlForce",bool_t,0,"Activate force control",False)
gen.add("targetForce",double_t,0,"Target force to apply [N]",0.0,0.0,20.0)
gen.add("k1",double_t,0,"k1",10,0.0,50.0)
gen.add("k2",double_t,0,"k2",1,0.0,10.0)
gen.add("minFc",double_t,0,"min Fc",-15.0,-50.0,0.0)
gen.add("maxFc",double_t,0,"max Fc",30.0,0.0,50.0)
gen.add("linearVhLimit",double_t,0,"Linear human velocity limit [m/s]",0.3,0,0.6)
gen.add("angularVhLimit",double_t,0,"Angular human velocity limit [rad/s]",1.0,0,3)
gen.add("arbitrationLimit",double_t,0,"Aribitration Limit (max autonomy value)",0.8,0,1.0)
gen.add("agreementWeight",double_t,0,"Weight on agreement confidence between human and task",0.2,0,1)


exit(gen.generate(PACKAGE, "motion_force_control", "forceTaskSharedControl_params"))